pages/bill/[billNumber].js

```diff
@@ -0,0 +1,49 @@
+import { documents } from "../../components/MockAPIResponseDocuments"
+import { createPage } from "../../components/page"
+import { Button, Row } from 'react-bootstrap'
+import SearchBar from '../../components/SearchBar/SearchBar'
+import Bill from '../../components/Bill/Bill'
+import BillTestimonies from '../../components/BillTestimonies/BillTestimonies'
+import AddTestimony from '../../components/AddTestimony/AddTestimony'
+
+export const getStaticPaths = async () => {
```

Comment:
With getStaticPaths and getStaticProps, Next.js pre-renders the bill pages when we run `yarn export`, producing a separate html file for each of the ~8000 bills. When you run `yarn dev`, next generates these pages as you visit them so the build time isn't noticeable.

Instead, we'll want to fetch the list of bills and bill content in the browser (using hooks like the one in #201) and then render the appropriate UI once it's loaded ([client-side data fetching](https://nextjs.org/docs/basic-features/data-fetching/client-side)). In this way we can have a single bills page html file and a single bill detail page html file. To specify which bill to load on the bill page, we can use query parameters, so the bill page url becomes `/bill?id=H123`. You can access the parameter from the page component with `router = useRouter(); const id = router.query.id`

IMO the Next documentation is misleading. Dynamic routing is normally associated with dynamic content, but they use it to mean dynamic-until-build-time. Probably to get you to pay for their hosted services.